gpkitmodels/SP/aircraft/prop/propeller.py

```diff
@@ -0,0 +1,127 @@
+" propeller model "
+from numpy import pi
+from gpkit import Model, Variable,Vectorize,parse_variables, SignomialsEnabled, SignomialEquality
+from gpkit.constraints.tight import Tight as TCS
+from gpfit.fit_constraintset import XfoilFit
+import os
+import pandas as pd
+
+
+
+class Blade_Element_Performance(Model):
+    """ Single element of a propeller blade
+
+    Variables
+    ---------
+    dT                      [lbf]       thrust
+    eta_i                   [-]         local induced efficiency
+    dQ                      [N*m]       torque
+    omega                   [rpm]       propeller rotation rate 
+    Wa                      [m/s]       Axial total relative velocity
+    Wt                      [m/s]       Tangential total relative velocity
+    Wr                      [m/s]       Total relative velocity
+    va                      [m/s]       Axial induced velocity
+    vt                      [m/s]       Tangential induced velocity
+    G                       [m^2/s]     Circulation
+    cl                     [-]          local lift coefficient
+    cd                      [-]         local drag coefficient
+    B           2           [-]         number of blades
+    r                       [m]         local radius
+    lam_w                   [-]         advance ratio
+    eps                     [-]         blade efficiency
+    AR_b                   [-]          blade aspect ratio
+    AR_b_max    50          [-]         max blade aspect ratio
+    Re                      [-]         blade reynolds number
+    f                       [-]         intermediate tip loss variable
+    F                       [-]         Prandtl tip loss factor
+    cl_max      .6         [-]          max airfoil cl
+    dr                      [m]         length of blade element
+    M                       [-]         Mach number
+    a           295         [m/s]       Speed of sound at altitude
+
+    """
+
+    def setup(self,static,  state):
+        exec parse_variables(Blade_Element_Performance.__doc__)
+
+        V       = state.V
+        rho     = state.rho
+        R       = static.R
+        mu      = state.mu
+        path = os.path.dirname(__file__)
+        fd = pd.read_csv(path + os.sep + "dae51_fitdata.csv").to_dict(
+            orient="records")[0]
+        c = static.c
+        constraints = [TCS([Wa>=V + va]),
+                        TCS([Wt + vt<=omega*r]),
+                        TCS([G == (1./2.)*Wr*c*cl]),
+                        F == (2./pi)*(1.01116*f**.0379556)**(1./.1), #This is the GP fit of arccos(exp(-f))
```

Comment:
Is there a reason you didn't make this a fit constraint set?

Reply:
also why is it raised to 1./1.

Reply:
simplicity of the fit; it's raised to 1/.1 (otherwise known as 10)